spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/StreamDeployer.java

```diff
@@ -80,4 +114,70 @@ protected String getDeploymentPath(StreamDefinition definition) {
 		return Paths.build(Paths.STREAM_DEPLOYMENTS, definition.getName());
 	}
 
+	/**
+	 * The migration code to run against existing stream definitions that don't have module definitions set already.
+	 * The following method will update the module definitions for those stream definitions.
+	 * See https://jira.spring.io/browse/XD-2854
+	 */
+	@PostConstruct
+	private void updateModuleDefinitions() {
+		if (this.zkConnection.getClient() != null) {
+			try {
+				CuratorFramework client = this.zkConnection.getClient();
+				if (client.checkExists().forPath(Paths.STREAMS) != null) {
+					Iterable<StreamDefinition> streamDefinitions = findAll();
+					for (StreamDefinition definition : streamDefinitions) {
+						setModuleDefinitions(client, definition);
+					}
+				}
+			}
+			catch (Exception e) {
+				logger.error("Exception migrating stream definitions. This migration is done when the existing " +
+						"stream definitions that don't have module definitions set. " + e);
```

Comment:
Pass the exception as a parameter to the logger so that we don't lose the stack trace.

As in, do `", e);` instead of `" + e);`.